pyomo/util/subsystems.py

```diff
@@ -0,0 +1,195 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+from pyomo.core.base.PyomoModel import ConcreteModel
+from pyomo.core.base.reference import Reference
+from pyomo.core.expr.visitor import identify_variables
+from pyomo.common.collections import ComponentSet, ComponentMap
+
+
+def create_subsystem_block(constraints, variables=None, include_fixed=False):
```

Comment:
Out of curiosity, what is the value of the `include_fixed` parameter?

Reply:
If a fixed variable may be unfixed at some point, then it needs to be attached to the subsystem block in order for that block to be a "valid model" according to the nl writer. We could want to solve this block after inputs are unfixed, perform a Dulmage-Mendelsohn decomposition, etc. I don't have a particular use case in mind, but it is conceivably possible that this would be useful/necessary at some point.